lib/ansible/config/base.yml

```diff
@@ -1441,15 +1441,17 @@ INTERPRETER_PYTHON_FALLBACK:
   version_added: "2.8"
 TRANSFORM_INVALID_GROUP_CHARS:
   name: Transform invalid characters in group names
-  default: False
+  default: 'never'
   description:
     - Make ansible transform invalid characters in group names supplied by inventory sources.
-    - If 'false' it will allow for the group name but warn about the issue.
-    - When 'true' it will replace any invalid charachters with '_' (underscore).
+    - If 'never' it will allow for the group name but warn about the issue.
+    - When 'always' it will replace any invalid charachters with '_' (underscore) and warn the user
+    - When 'silently', it does the same as 'always' sans the warnings.
```

Comment:
Please add `ignore` as well!

Reply:
still not convinced that should be an option

Reply:
So you allow us to keep the dashes, but only if we accept that every time we run a playbook, we get a ton of deprecation warnings with no way to turn them off (except by turning *all* deprecation warnings off)? I'm currently getting a screen full of deprecation warnings at the beginning of every playbook run with `devel`. That's really annoying.

Reply:
i need to separate the deprecation warning with the per instance warning, deprecation will only exist for 2 versions till we flip the default

Reply:
Except for people who explicitly set the setting to the current default. Anyway, would it be possible to at least reduce the amounts of warnings, by printing one warning in total (or one per inventory plugin)? I guess it should contain a list of all invalid group names, which can be a very long warning if you have a lot of such groups, but at least it will take a lot of less screen space than the current huge list of warnings.

Reply:
try now